build_wheels.py

```diff
@@ -0,0 +1,195 @@
+import json
+import os
+import re
+from typing import List
+from typing import Union
+
+import requests
+from packaging.requirements import InvalidRequirement
+from packaging.requirements import Requirement
+
+# GLOBAL VARIABLES
+# URL to fetch IDF branches from
+IDF_BRANCHES_URL = 'https://api.github.com/repos/espressif/esp-idf/branches?protected=true&per_page=100'
+# URL to download constraints file from (vX.Y.txt part is auto-completed)
+IDF_CONSTRAINTS_URL = 'https://dl.espressif.com/dl/esp-idf/espidf.constraints.'
+# URL for IDF 'resources' root directory for requirements paths
+IDF_RESOURCES_URL = 'https://raw.githubusercontent.com/espressif/esp-idf/'
+
+# Minimal IDF release version to take requirements from (v{MAJOR}.{MINOR})
+# Requirements from all release branches and master equal or above this will be considered
+# Specified in Github variables
+MIN_IDF_MAJOR_VERSION: int = int(os.environ.get('MIN_IDF_MAJOR_VERSION', '5'))
+MIN_IDF_MINOR_VERSION: int = int(os.environ.get('MIN_IDF_MINOR_VERSION', '0'))
+
+# Constraints file version is automatically taken from branch name
+# For master branch it is taken constraints file incremented by 1 in IDF minor version
+# (e.g when last release is v5.2 the constrains file for master is v5.3)
+# When no such a file exists the last release constraints file is taken for master branch
+#
+# This feature can be overwritten by setting the AUTO_VERSION_CONSTRAINS_FILE to False
+# in Github variables or manual workflow
+# then the value in MASTER_CONSTRAIN_FILE is taken
+AUTO_VERSION_CONSTRAINS_FILE: bool = bool(os.environ.get('AUTO_VERSION_CONSTRAINS_FILE', 'True'))
```

Comment:
@jakub-kocka  I don't think it's going to work the way you probably think...

If the string `False` is passed as value of `AUTO_VERSION_CONSTRAINS_FILE`, it will be bool True (a non-empty string is True).

Check the lib `distutils.util -> strtobool` (up to Python 3.10) or `distlib.util -> strtobool` (python 3.10 and newer, this one needs to be installed `pip install distlib`)

Alternatively you can use 1/0 as switch or do something like this:

```py
AUTO_VERSION_CONSTRAINS_FILE: bool = os.environ.get('AUTO_VERSION_CONSTRAINS_FILE', 'True').lower() == 'true'
```

Reply:
Thank you, @tomassebestik. That is right.